extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java

```diff
@@ -139,5 +313,236 @@ public void registerSwaggerUiServletExtension(SwaggerUiRecorder recorder,
+        Optional<String[]> requestCurlOptions;
+
+        /**
+         * MUST be a function. Function to intercept remote definition, "Try it out", and OAuth 2.0 responses.
+         * Accepts one argument responseInterceptor(response) and must return the modified response, or a Promise that resolves
+         * to the modified response.
+         */
+        @ConfigItem(name = "responseInterceptor")
+        Optional<String> responseInterceptor;
+
+        /**
+         * If set to true, uses the mutated request returned from a requestInterceptor to produce the curl command in the UI,
+         * otherwise the request before the requestInterceptor was applied is used.
+         */
+        @ConfigItem(name = "showMutatedRequest")
+        Optional<Boolean> showMutatedRequest;
+
+        /**
+         * List of HTTP methods that have the "Try it out" feature enabled.
+         * An empty array disables "Try it out" for all operations. This does not filter the operations from the display.
+         */
+        @ConfigItem(name = "supportedSubmitMethods")
+        Optional<HttpMethod[]> supportedSubmitMethods;
+
+        /**
+         * By default, Swagger UI attempts to validate specs against swagger.io's online validator.
+         * You can use this parameter to set a different validator URL, for example for locally deployed validators (Validator
+         * Badge).
+         * Setting it to either none, 127.0.0.1 or localhost will disable validation.
+         */
+        @ConfigItem(name = "validatorUrl")
+        Optional<String> validatorUrl;
+
+        /**
+         * If set to true, enables passing credentials, as defined in the Fetch standard, in CORS requests that are sent by the
+         * browser.
+         */
+        @ConfigItem(name = "withCredentials")
+        Optional<Boolean> withCredentials;
+
+        /**
+         * Function to set default values to each property in model. Accepts one argument modelPropertyMacro(property), property
+         * is immutable
+         */
+        @ConfigItem(name = "modelPropertyMacro")
+        Optional<String> modelPropertyMacro;
+
+        /**
+         * Function to set default value to parameters. Accepts two arguments parameterMacro(operation, parameter).
+         * Operation and parameter are objects passed for context, both remain immutable
+         */
+        @ConfigItem(name = "parameterMacro")
+        Optional<String> parameterMacro;
+
+        /**
+         * If set to true, it persists authorization data and it would not be lost on browser close/refresh
+         */
+        @ConfigItem(name = "persistAuthorization")
+        Optional<Boolean> persistAuthorization;
+
+        /**
+         * The name of a component available via the plugin system to use as the top-level layout for Swagger UI.
+         */
+        @ConfigItem
+        Optional<String> layout;
+
+        /**
+         * An array of plugin functions to use in Swagger UI.
```

Comment:
I think I would have the documentation say `A list of`.

Reply:
Happy to change that (the documentation is directly from the swagger-ui documentation) and it's really javascript (hence array)